kivy/resources.py

```diff
@@ -42,23 +43,42 @@
     resource_paths += [join(dirname(sys.argv[0]), 'YourApp')]
 resource_paths += [dirname(kivy.__file__), join(kivy_data_dir, '..')]
 
+Cache.register('kv.resourcefind', timeout=60)
 
-def resource_find(filename):
+
+def resource_find(filename, use_cache=True):
     '''Search for a resource in the list of paths.
     Use resource_add_path to add a custom path to the search.
+    By default, results are cached for 60 seconds.
+    This can be disabled using use_cache=False.
+
+    .. versionchanged:: 2.1.0
+        `use_cache` parameter added and made True by default.
     '''
     if not filename:
         return
+    found_filename = None
+    if use_cache:
+        found_filename = Cache.get('kv.resourcefind', filename)
+        if found_filename:
+            return found_filename
     if filename[:8] == 'atlas://':
-        return filename
-    if exists(abspath(filename)):
-        return abspath(filename)
-    for path in reversed(resource_paths):
-        output = abspath(join(path, filename))
-        if exists(output):
-            return output
-    if filename[:5] == 'data:':
-        return filename
+        found_filename = filename
+    else:
+        abspath_filename = abspath(filename)
+        if exists(abspath_filename):
+            found_filename = abspath(filename)
+        else:
+            for path in reversed(resource_paths):
+                abspath_filename = abspath(join(path, filename))
+                if exists(abspath_filename):
+                    found_filename = abspath_filename
+                    break
+        if not found_filename and filename.startswith("data:"):
+            found_filename = filename
+    if use_cache:
```

Comment:
```suggestion
    if use_cache and found_filename:
```

Because this can happen e.g. when `not found_filename and not filename.startswith("data:")`.

Reply:
fixed, and will look into tests

Reply:
tests added